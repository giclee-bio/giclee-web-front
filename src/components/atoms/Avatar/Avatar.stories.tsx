import { Meta, StoryObj } from '@storybook/react'

import Avatar from './Avatar'

const meta: Meta<typeof Avatar> = {}

export default meta
type Story = StoryObj<typeof Avatar>

export const Index: Story = {
  args: {
    alt: '',
    size: 'medium',
    src: '/nacal.png',
  },
}
