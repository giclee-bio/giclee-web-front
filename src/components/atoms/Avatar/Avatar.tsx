import classNames from 'classnames'
import Image from 'next/image'

type Props = {
  alt: string
  size: 'small' | 'medium' | 'large'
  src: string
}

const Avatar: React.FC<Props> = ({ alt, size, src }) => {
  let classes = classNames('rounded-full')
  let imageSize

  switch (size) {
    case 'small':
      classes = classNames(classes, '')
      imageSize = 40
      break
    case 'medium':
      classes = classNames(classes, '')
      imageSize = 80
      break
    case 'large':
      classes = classNames(classes, '')
      imageSize = 120
      break
    default:
      classes = classNames(classes, '')
  }

  return <Image alt={alt} className={classes} height={imageSize} src={src} width={imageSize} />
}

export default Avatar
