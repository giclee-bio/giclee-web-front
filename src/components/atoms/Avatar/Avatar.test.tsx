import { render } from '@testing-library/react'

import Avatar from './Avatar'

describe('Avatar', () => {
  it('snapshot', async () => {
    const { asFragment } = render(<Avatar alt='aaa' size='medium' src='/' />, {})
    expect(asFragment()).toMatchSnapshot()
  })
})
