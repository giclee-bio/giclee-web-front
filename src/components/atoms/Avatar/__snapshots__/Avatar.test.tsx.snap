// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar snapshot 1`] = `
<DocumentFragment>
  <img
    alt="aaa"
    class="rounded-full"
    data-nimg="1"
    decoding="async"
    height="80"
    loading="lazy"
    src="/_next/image?url=%2F&w=256&q=75"
    srcset="/_next/image?url=%2F&w=96&q=75 1x, /_next/image?url=%2F&w=256&q=75 2x"
    style="color: transparent;"
    width="80"
  />
</DocumentFragment>
`;
